get-started: initial refactoring

jorgeorpinel opened the pull request:
- Closes #747 (dedicated landing page for docs)
- Addresses part of #1035 (merge get-started and tutorials sections)

jorgeorpinel commented:
@shcheklein so far this only implements #747 but probably a little hacky – see `getItemByPath` in src/utils/sidebar.js and `SidebarMenu.timeout` in src/components/Documentation/SidebarMenu/index.js. Can you guys look at it and lmk what a better way to implement this could be? Cc @fabiosantoscode 

Also if this should be a PR of it's own I can separate it from the remaining changes (Get Started related).

shcheklein commented:
Looks good to me! We should spend a bit of time to think about the landing page content (check other projects). Also, see this discussion that is relevant to your changes in the sidebar.json -  https://github.com/iterative/dvc.org/pull/1025#commitcomment-37819121 ... cc @iAdramelk - if we can merge this one before Gatsby migration we might just disallow having any entry w/o index page? But to be honest I would still do redirects if file is missing, and we should be doing this on the middleware level automatically (not relying on sidebar.json).

shcheklein commented:
another point - we need to introduce some special "Home" entry to the sidebar?

jorgeorpinel commented:
> - [ ] We should spend a bit of time to think about the landing page content (check other projects)

Working on that ⏳ 

> we need to introduce some special "Home" entry to the sidebar?

I don't think so, we have the DOC link in the top menu. But if you prefer that then sure, that wouldn't require the js code changes I did.

> if we can merge this one before Gatsby migration

If you want to merge this PR soon, please approve it and I'll limit it to this first step for now.

shcheklein commented:
> I don't think so, we have the DOC link in the top menu. But if you prefer that then sure, that wouldn't require the js code changes I did.

yeah, I think it's good to clearly see the structure and being able to return to the page - here is an example https://www.algolia.com/doc/ .... we might even want to use an icon and separate it a bit from the other menu items, etc - need to think a bit about UX

> if we can merge this one before Gatsby migration

I would confirm with @iAdramelk first.

jorgeorpinel commented:
> think about the landing page content

Updated

> I think it's good to clearly see the structure and being able to return to the page

OK, done.

jorgeorpinel commented:
> I would confirm with @iAdramelk first.

Well, too late! Haha. I've merged get-started and tutorials into [/doc/learn](https://dvc-landing-refactor-ge-nouj86.herokuapp.com/doc/learn) now (so this PR has become big because of all the link renaming). What do you think?

shcheklein commented:
I think it can be very hard to merge both now. I would focus on the content for now, not significant structural changes.

shcheklein commented:
Or at least, we'll need to merge Gatsby migration first and then deal with this here.

shcheklein commented:
<img width="287" alt="Screen Shot 2020-03-15 at 8 39 13 PM" src="https://user-images.githubusercontent.com/3659196/76721349-0f705300-66fd-11ea-8566-f9b3bd63ad6b.png">


we should remove triangle, try to use icon and separate it from other items I think

shcheklein commented:
https://dvc-landing-refactor-ge-nouj86.herokuapp.com/doc/Home - url should be lower case

shcheklein commented:
content for the home page - probably should provide some good entry point and navigation for other sections

shcheklein commented:
also, `learn` - name is too general now, what is the plan for this section?

jorgeorpinel commented:
> we should remove triangle

Extracted to #1053

> learn - name is too general now, what is the plan for this section?

"Learn" is a common name for this kind of section with all the guides, tutorials, etc e.g. https://nextjs.org/learn 

---

I addressed your other couple comments @shcheklein.

jorgeorpinel commented:
I'm gonna limit the scope of this PR to what's done now. It's too big already. Fortunately its mostly due to renaming /doc/tutorials -> /doc/learn

jorgeorpinel commented:
> I would focus on the content for now, not significant structural changes

I thought structural changes for Get Started was the priority 😅 (both this one and then #599 which I've been working on as well). Should I postpone this work for now and focus on another high priority that is more content-related e.g. redesign the older use cases?

shcheklein commented:
@jorgeorpinel 

>  I thought structural changes for Get Started was the priority

that's true, bu to my mind the biggest potion of this is content - examples, preparing repos, writing text, etc - more than moving sections around.

jorgeorpinel commented:
> From https://github.com/iterative/dvc.org/issues/599#issuecomment-599883773
> ... Learn is not even part of Docs. Mostly because it's too generic. Second, it looks like the whole Learn section for Next is actually one single interactive tutorial

OK, I reverted back to /doc/tutorials but I still need to finish merging all the conflicts from the Gatsby migration... ⏳

jorgeorpinel commented:
Merged. https://dvc-landing-refactor-ge-nouj86.herokuapp.com/doc is running on Gatsby now.

shcheklein commented on redirects-list.json:
```diff
@@ -6,12 +6,13 @@
   "^/(?:help|chat)/?$                       https://discordapp.com/invite/dvwXA2N 303",
   "^/(?:docs|documentation)(/.*)?$          /doc$1",
-  "^/doc/commands-reference(/.*)?$          /doc/command-reference$1",
+  "^/doc/?$                                 /doc/tutorials/get-started 307",
```
should it be home now?

jorgeorpinel replied:
Yes, I also noticed this earlier, it's fixed. Thanks

shcheklein commented on redirects-list.json:
```diff
@@ -6,12 +6,13 @@
-  "^/doc/?$                                 /doc/get-started 307",
+  "^/doc/user-guide/contributing/?$         /doc/user-guide/contributing/core 307",
   "^/doc/understanding-dvc/?$               /doc/understanding-dvc/collaboration-issues 307",
-  "^/doc/changelog/?$                       /doc/changelog/0.18 307",
```
I really want to get rid of changelog, make it an external link ... again a special case like a Home button ... we spend time on it again and again

jorgeorpinel replied:
I thought it was an external link already before? I didn't introduce this in the Gatsby migration, just moved the order of some redirects.

shcheklein commented on src/components/Community/data.json:
```diff
@@ -28,13 +28,13 @@
     {
       "url": "/doc/tutorials",
-      "title": "Tutorials",
+      "title": "DVC Tutorials",
```
why DVC Tutorials?

jorgeorpinel replied:
I was changing just back from "Learn DVC", didn't remember it was just "Tutorials" before. Reverted.

jorgeorpinel commented:
Let's try to address the remaining refactoring in #1074 (or further PRs)? Any problems specific to these nav order changes I can still fix here, of course.
